Tests/MessagingPush/MessagingPushTest.swift

```diff
@@ -18,12 +18,36 @@ class MessagingPushTest: UnitTest {
         httpClientMock = HttpClientMock()
         identifyRepositoryMock = IdentifyRepositoryMock()
 
-        mockCustomerIO = CustomerIO(credentialsStore: nil, sdkConfig: SdkConfig(), identifyRepository: identifyRepositoryMock,
-                                keyValueStorage: nil)
+        mockCustomerIO = CustomerIO(credentialsStore: SdkCredentialsStoreMock(), sdkConfig: SdkConfig(), identifyRepository: identifyRepositoryMock, keyValueStorage: nil)
         
-        mockCustomerIO.setCredentials(siteId: String.random, apiKey: String.random, region: Region.EU)
+        mockCustomerIO.credentials = SdkCredentials(siteId: String.random,
+                                         apiKey: String.random,
+                                         region: Region.EU)
         
-        messagingPush = MessagingPush(customerIO: mockCustomerIO, httpClient: httpClientMock, keyValueStorage: DITracking.shared.keyValueStorage, jsonAdapter: jsonAdapter)
+        messagingPush = MessagingPush(customerIO: mockCustomerIO, httpClient: httpClientMock, jsonAdapter: jsonAdapter)
+    }
+    
+    private func pushSetup() -> MessagingPush{
+        let identifyRepository = CIOIdentifyRepository(httpClient: httpClientMock, keyValueStorage: DITracking.shared.keyValueStorage, jsonAdapter: jsonAdapter, siteId: String.random)
+        let cio = CustomerIO(credentialsStore: SdkCredentialsStoreMock(), sdkConfig: SdkConfig(), identifyRepository: identifyRepository, keyValueStorage: nil)
+        
+        cio.credentials = SdkCredentials(siteId: String.random,
+                                         apiKey: String.random,
+                                         region: Region.EU)
+    
+        httpClientMock.requestClosure = { params, onComplete in
+            onComplete(Result.success(Data()))
+        }
+        
+        let identifier: String? = String.random
+
+    
+        cio.identify(identifier: identifier!) { result in
+            guard case .success = result else {  return XCTFail() }
+            XCTAssertEqual(cio.identifier, identifier)
+        }
+        
+        return MessagingPush(customerIO: cio, httpClient: httpClientMock, jsonAdapter: jsonAdapter)
```

Comment:
This will be improved when I perform the refactors that I have mentioned a couple of times. Dependency injection is not not really setup the best at the moment which makes creating mocks and instances harder. This is a good workaround for now.